hudi-common/src/main/avro/HoodieMetadata.avsc

```diff
@@ -23,7 +23,10 @@
     "fields": [
         {
             "name": "key",
-            "type": "string"
+            "type": [
```

Comment:
null is not needed here as instead of nulling this field you can store an empty string.

The type=null signifies that this field may be missing from the record.

Reply:
Good point @prashantwason @nsivabalan. Will revert to string, and have the metadata serialization put the empty string.